lib/mpl_toolkits/tests/test_axisartist_grid_helper_curvelinear.py

```diff
@@ -0,0 +1,219 @@
+
+
+@image_comparison(baseline_images=['custom_transform'],
+                  extensions=['png'], style='default', tol=0.03)
+def test_custom_transform():
+    class MyTransform(Transform):
+        input_dims = 2
+        output_dims = 2
+        is_separable = False
+
+        def __init__(self, resolution):
+            """
+            Resolution is the number of steps to interpolate between each input
+            line segment to approximate its path in transformed space.
+            """
+            Transform.__init__(self)
+            self._resolution = resolution
+
+        def transform(self, ll):
+            x = ll[:, 0:1]
+            y = ll[:, 1:2]
+
+            return np.concatenate((x, y - x), 1)
+
+        transform_non_affine = transform
+
+        def transform_path(self, path):
+            vertices = path.vertices
+            ipath = path.interpolated(self._resolution)
+            return Path(self.transform(ipath.vertices), ipath.codes)
+
+        transform_path_non_affine = transform_path
+
+        def inverted(self):
+            return MyTransformInv(self._resolution)
+
+    class MyTransformInv(Transform):
+        input_dims = 2
+        output_dims = 2
+        is_separable = False
+
+        def __init__(self, resolution):
+            Transform.__init__(self)
+            self._resolution = resolution
+
+        def transform(self, ll):
+            x = ll[:, 0:1]
+            y = ll[:, 1:2]
+
+            return np.concatenate((x, y+x), 1)
+
+        def inverted(self):
+            return MyTransform(self._resolution)
+
+    fig = plt.figure()
+
+    SubplotHost = host_subplot_class_factory(Axes)
+
+    tr = MyTransform(1)
+    grid_helper = GridHelperCurveLinear(tr)
+    ax1 = SubplotHost(fig, 1, 1, 1, grid_helper=grid_helper)
+    fig.add_subplot(ax1)
+
+    ax2 = ParasiteAxesAuxTrans(ax1, tr, "equal")
+    ax1.parasites.append(ax2)
+    ax2.plot([3, 6], [5.0, 10.])
+
+    ax1.set_aspect(1.)
+    ax1.set_xlim(0, 10)
+    ax1.set_ylim(0, 10)
+
+    ax1.grid(True)
+
+
+@image_comparison(baseline_images=['polar_box'],
+                  extensions=['png'], style='default', tol=0.03)
+def test_polar_box():
+    fig = plt.figure(figsize=(5, 5))
+
+    # PolarAxes.PolarTransform takes radian. However, we want our coordinate
+    # system in degree
+    tr = Affine2D().scale(np.pi / 180., 1.) + PolarAxes.PolarTransform()
+
+    # polar projection, which involves cycle, and also has limits in
+    # its coordinates, needs a special method to find the extremes
+    # (min, max of the coordinate within the view).
+    extreme_finder = angle_helper.ExtremeFinderCycle(20, 20,
+                                                     lon_cycle=360,
+                                                     lat_cycle=None,
+                                                     lon_minmax=None,
+                                                     lat_minmax=(0, np.inf))
+
+    grid_locator1 = angle_helper.LocatorDMS(12)
```

Comment:
Just to be sure: switching from `5` to `12` was done on purpose?

Reply:
This is probably from `curvelinear_test3`; there's also `curvelinear_test2` which I think I squashed into one test because they're pretty much the same.